src/Analyzers/Core/Analyzers/FileHeaders/AbstractFileHeaderDiagnosticAnalyzer.cs

```diff
@@ -63,7 +63,10 @@ private void HandleSyntaxTree(SyntaxTreeAnalysisContext context)
             }
 
             var expectedFileHeader = fileHeaderTemplate.Replace("{fileName}", Path.GetFileName(tree.FilePath));
-            if (!CompareCopyrightText(expectedFileHeader, fileHeader.CopyrightText))
+            if (!CompareCopyrightText(expectedFileHeader, fileHeader.CopyrightText) &&
+                // Compare the current fileHeader with the expected file header, assuming the expected file header is written as a comment
+                // e.g. file_header_template = /* Copyright */
+                !CompareCopyrightText(expectedFileHeader, tree.GetText(context.CancellationToken).GetSubText(fileHeader.HeaderSpan).ToString()))
             {
                 context.ReportDiagnostic(Diagnostic.Create(InvalidHeaderDescriptor, fileHeader.GetLocation(tree)));
                 return;
```

Comment:
you can just break this out into two statements :)  but i'm not going to push heavily on this.

Reply:
Done b4fec156.